### PR TITLE
Generic docker Workflow

### PR DESCRIPTION
**Pull request by SebastianThomas**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ If your project **does not** use tag based workflows (e.g. using `@main`) and so
 
 **Quick fix:** Update all usages of `some-workflow@main` to `some-workflow@v1`.
 
+## v2.1
```
Comment (dsparber) [inline CHANGELOG.md:7]:
```suggestion
## v3 (upcoming)
```

File: .github/workflows/generic_docker_image.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+name: Build and Push Docker Images on release/
+on:
+    workflow_call:
+        inputs:
+            ref_name:
```
Comment (dsparber) [inline .github/workflows/generic_docker_image.yml:5]: Unused input variable?

File: .github/workflows/generic_docker_image.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    workflow_call:
+        inputs:
+            image_name:
+                required: true
+                type: string
+                description: name of the image to build
+            tag_name:
+                required: true
+                type: string
+                description: name of the tag
```
Comment (kistlers) [inline .github/workflows/generic_docker_image.yml:12]:
```suggestion
                description: name of the tag to build
```

**Review by kistlers — APPROVED** [review]

**Review by kistlers — CHANGES_REQUESTED** [review]
See comments

**Review by kistlers — APPROVED** [review]

**Pull request merged**